### PR TITLE
ARROW-367: converter json <=> Arrow file format for Integration tests

### PR DESCRIPTION
**Pull request by julienledem**

### CODE REVIEW

**Comment by julienledem** [comment]
@wesm here is a tool to convert from arrow to json and back as well as validate an arrow file against a json one.

**Comment by wesm** [comment]
fantastic, will review today. made good progress yesterday on the JSON file format in C++, so will write an equivalent tool

File: java/tools/src/main/java/org/apache/arrow/tools/Integration.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+      FieldVector arrowVector = arrowVectors.get(i);
+      FieldVector jsonVector = jsonVectors.get(i);
+      int valueCount = arrowVector.getAccessor().getValueCount();
+      if (valueCount != jsonVector.getAccessor().getValueCount()) {
+        throw new IllegalArgumentException("Different value count for field " + field + " : " + valueCount + " != " + jsonVector.getAccessor().getValueCount());
+      }
+      for (int j = 0; j < valueCount; j++) {
+        Object arrow = arrowVector.getAccessor().getObject(j);
+        Object json = jsonVector.getAccessor().getObject(j);
+        if (!Objects.equal(arrow, json)) {
```
Comment (wesm) [inline java/tools/src/main/java/org/apache/arrow/tools/Integration.java]: This works for nested types and nulls?

File: java/tools/src/test/java/org/apache/arrow/tools/ArrowFileTestFixtures.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import org.apache.arrow.vector.file.ArrowReader;
+import org.apache.arrow.vector.file.ArrowWriter;
+import org.apache.arrow.vector.schema.ArrowRecordBatch;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.junit.Assert;
+
+public class ArrowFileTestFixtures {
+  static final int COUNT = 10;
+
+  static void writeData(int count, MapVector parent) {
```
Comment (wesm) [inline java/tools/src/test/java/org/apache/arrow/tools/ArrowFileTestFixtures.java]: Come to think of it, many of these test cases could use some null values

**Review by wesm — COMMENTED** [review]
+1 LGTM modulo minor comments. We'll be sure to include nulls in the integration test datasets

File: java/tools/src/main/java/org/apache/arrow/tools/Integration.java (hunk 1 above)
Comment (julienledem) [inline java/tools/src/main/java/org/apache/arrow/tools/Integration.java]: yes.
Objects.equals takes care of null.
getObject(index) materializes nested types and lists.

File: java/tools/src/test/java/org/apache/arrow/tools/ArrowFileTestFixtures.java (hunk 2 above)
Comment (julienledem) [inline java/tools/src/test/java/org/apache/arrow/tools/ArrowFileTestFixtures.java]: good point

**Comment by wesm** [comment]
+1

File: java/tools/src/test/java/org/apache/arrow/tools/ArrowFileTestFixtures.java (hunk 2 above)
Comment (julienledem) [inline java/tools/src/test/java/org/apache/arrow/tools/ArrowFileTestFixtures.java]: sorry, I was busy this week, but I will add more tests as suggested.

**Pull request closed without merging**